### ISSUE TITLE
Inconsistent display of join date (also breaking localisations) 

### ISSUE THREAD

**Issue by Olf0** [mastodon/mastodon#16661]
### Expected behaviour

Join date shall always displayed as a date, e.g. 4 July 2021
I.e., in `date`'s syntax (and descending personal preference):
`%d %B %Y` (e.g., 4 July 2021) *or* `%x` (i.e., the localised variant of `%D`) *or* `%F` (ISO-format)

### Actual behaviour

Join date is displayed as an abbreviated month, e.g. Jul 2021
I.e., in `date`'s syntax: `%b %Y`

### Steps to reproduce the problem

1. Look at the "join date" on the home-page of a Mastodon-account in the right column, below the account description.
   Examples:
   a. https://mastodon.social/@Gargron
   b. https://social.bund.de/@bfdi
   See that it uses an abbreviated month (to three letters, e.g. Jul 2021), instead of a full date (including the day, e.g. 4 July 2021).
2. In contrast to that a full date is displayed at exactly the same spot, when looking at e.g.
   c. https://social.bund.de/web/accounts/8666
   One needs to be logged in to the corresponding server to see that, but the main point is: The full date is already used at this spot, when using a different URL format!
3. While the English string ("Joined") is adapted for displaying the join date as abbreviated month plus year (see example "a." in 1.), it fails when displaying a Mastodon home-page in the URL-format of "c." above (in 3.): It shall be rectified to "Joined on <date>", plus using the full date (e.g., 4 July 2021).
4. In contrast to the original English string, many localised strings are correctly phrased for using the full date (e.g. the German localisation "Beigetreten am") but fail due to the currently used abbreviated month (see example "b." in 1.)!
   But these localised strings are absolutely fine when displaying a Mastodon home-page in the URL-format of "c." above (in 3.)
5. As discussed in comments of the merge requests referenced below, it does not make sense to include the time of the join action (if available): The full date is sufficient.
   On the other hand it also does not make sense to save a few characters by showing only the abbreviated month (plus year): There is sufficient space for a full date.

TL;DR:
1. Please use the full date (e.g., 4 July 2021) for the "join date" at the home-page of a Mastodon-account in the right column, below the self description.
2. Consequently adapt the English string for this to "Joined on <date>" (from "Joined").
3. Check which localised strings were adapted to the use of abbreviated month (and alter them).
   Note that many do already have the correct grammar for addressing a full date, hence must not be changed.

Related, merged MRs: #16169 and #16186
(Related, closed MR: #1653)
Maybe related issues: #16187, plus (unlikely) #5399 (and its duplicate #5791)

### Specifications

Observed with Mastodon server version 3.4.1

Irrelevant side note: Used web browser is Firefox 78.13.0esr (Linux-x86, 64 bit)

**Comment by Olf0** [comment]
@trwnh, why has this issue been "closed as completed"?  Is there a commit or PR which was missed to be referenced here?
